### ISSUE TITLE
Port to Linux and MacOs

### ISSUE THREAD

**Issue by julienblitte** [julienblitte/UniversalScanner#1]
Study port to Linux and MacOs

**Comment by JohnTar22** [comment]
i'd really like to see a linux port! :)

**Comment by alkaphreak** [comment]
Good job pal.

**Comment by julienblitte** [comment]
> i'd really like to see a linux port! :)

Yeap, I would like maybe first to migrate to another development language. Might be C or Rust.
I have already made a proof of concept working (in C) in command line for Linux for Dahua and HIK Vision.

> Good job pal.

:-)

**Comment by alkaphreak** [comment]
Rust would be a good choice.

**Comment by JohnTar22** [comment]
Thanks for the sarcasm, for the sake of others, I've managed to use mono to run it on Ubuntu, works pretty good.

**Comment by julienblitte** [comment]
> Thanks for the sarcasm

Misunderstanding, there is no sarcasm, I am really working on this topic and really think about porting this to C or Rust.

>for the sake of others, I've managed to use mono to run it on Ubuntu, works pretty good.

Great! Feel free to share any tips you found to make it works.

For now, I have done this piece of code, I share with you if it can help you also:
https://github.com/julienblitte/uscan

I made some progress on this, I will update you.

**Comment by JohnTar22** [comment]
Thanks for the answer, Iv'e tried uscan, looks good!, as i see there's still a big difference in the vendor support list between the versions, am i right?

**Comment by julienblitte** [comment]
Yes, I have put in the readme [the list of missing protocol](https://github.com/julienblitte/uscan#protocols-to-be-ported) compared to this **C#** program, as a to-do-list / reminder and there is still a lot to do. 

But I am still not sure to finish this, if I switch to **Rust** meanwhile... 

Next protocols to port would be probably **mDNS** and **GigEVision** as they are _generic protocols_.

**Comment by ramees-creator** [comment]
Dear Team ,

Can you make mac version ( Apple Silicon )
